### ISSUE TITLE
AVRO schema for Struct keys generated on table creation does not match the schema expected by inserts

### ISSUE THREAD

**Issue by Gerrrr** [confluentinc/ksql#8873]
**Describe the bug**

AVRO schema for Struct keys generated on table creation does not match the schema expected by inserts.

**To Reproduce**

~ksqlDB release 0.21.0 (the version bundled with Confluent Platform 7.0.1)~, also reproduces on latest `master`.

```sql
ksql> CREATE TABLE bar (id STRUCT<id BIGINT> PRIMARY KEY, name VARCHAR, fk BIGINT) WITH (kafka_topic='bar', format='AVRO', partitions=64);

 Message
---------------
 Table created
---------------
ksql> INSERT INTO bar (id, name, fk) VALUES (Struct(id := 0), 'name-0', 0);
```

**Expected behavior**

Insertion succeeds.

**Actual behaviour**

```
Failed to insert values into 'BAR'. Cannot INSERT VALUES into data source `BAR`. ksqlDB generated schema would overwrite existing key schema.
	Existing Schema: {"type":"record","name":"BarKey","namespace":"io.confluent.ksql.avro_schemas","fields":[{"name":"ID","type":["null","long"],"default":null}]}
	ksqlDB Generated: {"type":"record","name":"BarKey","namespace":"io.confluent.ksql.avro_schemas","fields":[{"name":"ID","type":["null","long"],"default":null}],"connect.name":"io.confluent.ksql.avro_schemas.BarKey"}
```

**Comment by agavra** [comment]
This is a duplicate of #7211 which I believe @lihaosky

**Comment by lihaosky** [comment]
@Gerrrr , FYI, I merged the fix for #7211

**Comment by Gerrrr** [comment]
Thank you @lihaosky ! It solved the issue!
